src/display-buffer.coffee

```diff
@@ -41,6 +41,12 @@ class DisplayBuffer extends Model
     @disposables.add @tokenizedBuffer.onDidChange @handleTokenizedBufferChange
     @disposables.add @buffer.onDidCreateMarker @handleBufferMarkerCreated
     @foldMarkerAttributes = Object.freeze({class: 'fold', displayBufferId: @id})
+    if @largeFileMode
+      @tokenizedBuffer.onDidLoad(=> @finalizeConstruction())
```

Comment:
I wonder if some day code path for large files will be good enough to use it unconditionally for both large and small files.